src/main/java/org/owasp/esapi/waf/internal/InterceptingHTTPServletRequest.java

```diff
@@ -171,18 +172,43 @@ public Enumeration getDictionaryParameterNames() {
 	private class RAFInputStream extends ServletInputStream {
 		
 		RandomAccessFile raf;
+		boolean isDone = false;
+		
 		
 		public RAFInputStream(RandomAccessFile raf) throws IOException {
 			this.raf = raf;
 			this.raf.seek(0);
 		}
 
 		public int read() throws IOException {
-			return raf.read();
+			int rval = raf.read();
+			
+			if (rval == -1) {
+				isDone = true;
+			}
+			
+			return rval;
 		}
 		
 		public synchronized void reset() throws IOException {
 			raf.seek(0);
+			isDone = false;
+		}
+
+		@Override
+		public boolean isFinished() {
+			return isDone;
+		}
+
+		@Override
+		public boolean isReady() {
+			// No way to verify that the next read would be non-blocking.
+			return false;
+		}
+
+		@Override
+		public void setReadListener(ReadListener readListener) {
```

Comment:
I'm wondering here if we shouldn't throw a not implemented exception?

Reply:
I waffled on this a bit.   Seeing as how this class is a private internal, I don't think it will make too much of a difference overall.
From what I recall, we never directly call this method and we do not expose the RAFInputStream to anyone but the InterceptingHTTPServletRequest (the outer class in this case).

If anyone ever finds that this method isn't doing what they expect, they're already in the baseline making updates.  Shouldn't be too hard for them to track down either way.

Reply:
This has been reverted while rolling back to javax.servlet-api 3.0.1
https://github.com/ESAPI/esapi-java-legacy/pull/683/commits/54c56d989393b1a0c02cb3ef9a428f5c11d7729f

Reply:
restored in update to 3.1.0
https://github.com/ESAPI/esapi-java-legacy/pull/683/commits/959d9e12817804a78e719b2dc42dc49194d43b41

Reply:
@xeno6696, how strongly do you feel about this exception?

Reply:
Not very.  It was definitely more of a suggestion.